### PR TITLE
[docs] Add a note about pull request sizes in the contributing guide

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
We're seeing bigger and bigger PRs. It would be good for the overall DX if contributors were aware of the effect of huge PRs on the whole PR process.

Reviewers will be able to reference this section when faced with a pull request that can be split into multiple ones. I will avoid discussions between the reviewer and the contributor about who is the laziest of the two (I've already witness this situation in another project haha)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -132,6 +132,31 @@ accepted. For example:
...
+* **Higher quality reviews**: It is much easier to spot a bug in a few lines than in 1000 lines.
+* **Faster overall review**: Reviewers, to approve a pull request, need to understand every line and understand how
+it fits into your overall change. They also need to go back and forth between files and functions to understand the flow
+of the code. This is exponentially hard as there are more lines in the code.
+* **Avoiding blocking changes that are valid**: In a huge pull request, it's likely that some changes are valid and some need to
+be reworked/discussed. If all the changes are in the same pull request, then the valid changes will be be blocked until
+all discussions have been resolved.
+* **Reducing the number of git conflicts**: Bigger pull request means slower reviews, thus means that the pull request will be open
+longuer and will have more git conflicts to be resolved before being merged.
+* **Parrallel processing**: All programmers like to parralelize. Well reviewers also like to parralelize code reviews to merge your code
```
Comment (JoeLoser) [inline CONTRIBUTING.md:152]:
```suggestion
* **Parallel processing**: All programmers like to parallelize. Well reviewers also like to parallelize code reviews to merge your code
```

**Review by JoeLoser — APPROVED** [review]
Thanks for writing this up!  Definitely agree we've been seeing some big ones lately which can slow down the queue overall right now with where our current team size is at.

**Comment by JoeLoser** [comment]
@gabrieldemarmiesse do you mind fixing up some markdownlint things for this contribution, please?  I just added the same markdownlint and config we use internally.  You can repro the errors I see internally with

```
./stdlib/scripts/markdownlint.sh CONTRIBUTING.md
```

I'll file a GitHub issue for us to run `markdownlint.sh` on all the markdown files in the public `mojo` repo too to make our lives easier.

**Comment by gabrieldemarmiesse** [comment]
@JoeLoser done, please merge https://github.com/modularml/mojo/pull/2717 before.

**Pull request closed without merging**